content/en/docs/Configuration/p8s-jaeger-grafana/prometheus.md

```diff
@@ -0,0 +1,183 @@
+
+- Appending `|istio_request_duration_milliseconds_*` to the `drop` regex above would drop all associated metrics and would prevent any request duration/response time reporting in Kiali.
+- Appending `|istio_request_duration_milliseconds_bucket` to the `drop` regex above, would prevent any request duration/response time percentile reporting in the Kiali metric charts or graph edge labels.
+
+
+### Scrape Interval
+
+The Prometheus `globalScrapeInterval` is an important configuration option[^2]. The scrape interval can have a significant effect on metrics collection overhead as it takes effort to pull all of those configured metrics and update the relevant time-series. And although it doesn't affect time-series cardinality, it does affect storage for the data-points, as well as having impact when computing query results (the more data-points, the more processing and aggregation).
+
+[^2]: Note that Prometheus can be configured such that individual scrape points can override the global setting, but Kiali is not currently concerned with this corner case.
+
+Users should think carefully about their configured scrape interval. Note that the Istio addon for prometheus configures it to 15s. This is great for demos but may be too frequent for production scenarios. The prometheus helm charts set a default of 1m, which is more reasonable for most installations, but may not be the desired frequency for any particular setup.
+
+The recommendation for Kiali is to set the longest interval possible, while still providing a useful granularity. The longer the interval, the less data points scraped, reducing processing, storage, and computational overhead. But, the impact on Kiali should be understood. First, it is important to realize that request rates (or byte rates, message rates, etc) require a minumum of two data points:
+
+`rate = (dp2 - dp1) / timePeriod`
+
+That means for Kiali to show anything useful in the graph, or anywhere rates are used (many places), the minimum duration must be `>= 2 x globalScrapeInterval`. Kiali will [eliminate invalid Duration options]({{< ref "/docs/faq/graph#scrapeduration" >}}) given the globalScrapeInterval.
+
+Kiali does a lot of aggregation and querying over time periods. As such, the number of data points will affect query performance, especially for larger time periods.
+
+For more information, see the [Prometheus documentation](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#configuration).
+
+
+### TSDB retention time
+
+The Prometheus `tsdbRetentionTime` is an important configuration option.  This has significant effect on metric storage, as Prometheus will keep each reported data-point for that period of time, performing compaction as needed. The larger the retention time the larger the required storage.  Also, Kiali will offer larger time periods, and if the amount of data is very large, queries against those large time periods may result in poor performance or timeouts.
```

Comment:
```suggestion
The Prometheus `tsdbRetentionTime` is an important configuration option.  This has significant effect on metric storage as Prometheus will keep each reported data-point for that period of time performing compaction as needed. The larger the retention time, the larger the required storage.  Also, Kiali will offer larger time periods and, if the amount of data is very large, queries against those large time periods may result in poor performance or timeouts.
```

Reply:
Thanks, I missed a comma there.  Also, I re-wrote the last sentence completely.